best-practices/check_deprecated_apis.yaml

```diff
@@ -0,0 +1,33 @@
+apiVersion: kyverno.io/v1
+kind: ClusterPolicy
+metadata:
+  name: check-deprecated-apis
+  annotations:
+    policies.kyverno.io/category: Best Practices
+    policies.kyverno.io/description: >-
+      Kubernetes APIs are sometimes deprecated and removed after a few releases. 
+      As a best practice older API versions should be replaced with newer versions.
+spec:
+  validationFailureAction: enforce
+  rules:
+  - name: validate-v1-22-removals
+    match:
+      resources:
+        kinds:
+        - admissionregistration.k8s.io/v1beta1/ValidatingWebhookConfiguration
+        - admissionregistration.k8s.io/v1beta1/MutatingWebhookConfiguration
+        - apiextensions.k8s.io/v1beta1/CustomResourceDefinition
+        - apiregistration.k8s.io/v1beta1/APIService
+        - authentication.k8s.io/v1beta1/TokenReview
+        - authorization.k8s.io/v1beta1/SubjectAccessReview
+        - authorization.k8s.io/v1beta1/LocalSubjectAccessReview
+        - authorization.k8s.io/v1beta1/SelfSubjectAccessReview 
```

Comment:
These are blocked in the default resource filters: `'[Event,*,*][*,kube-system,*][*,kube-public,*][*,kube-node-lease,*][Node,*,*][APIService,*,*][TokenReview,*,*][SubjectAccessReview,*,*][SelfSubjectAccessReview,*,*][*,kyverno,*] [Binding,*,*][Repl      icaSet,*,*][ReportChangeRequest,*,*][ClusterReportChangeRequest,*,*]'`

Reply:
You are, of course, correct, but in this case a policy which specifies these may still be useful in cases where users have gone and modified the ConfigMap after the fact (and possibly can't envision the future impact). If they remain filtered out in the ConfigMap, then this policy will obviously not be able to block them.

Reply:
Good point! Wondering if we need to add a warning about policies that use kinds that are filtered, as a Kyverno enhancement.

Reply:
That's a good idea!

Reply:
Added: https://github.com/kyverno/kyverno/issues/2163.